src/pynxtools_xps/vms/casa_data_model.py

```diff
@@ -86,25 +85,70 @@ def process_alignment(self, align_str: str):
         None.
 
         """
-        split_line = self._split_line(align_str)
+        split_line = self._split_line(calib_str)
 
-        align = CasaAlignment()
+        energy_calib = CasaEnergyCalibration()
 
         measured_index = split_line.index("M")
         aligned_index = split_line.index("A")
-        align.measured_energy = float(split_line[measured_index + 2])
-        align.aligned_energy = float(split_line[aligned_index + 2])
+        try:
+            energy_calib.measured_energies = [float(split_line[measured_index + 2])]
+        except ValueError:
+            energies = [
+                float(energy) for energy in split_line[measured_index + 2].split(",")
+            ]
+            energy_calib.measured_energies = energies
+            energy_calib.range_calibration = True
+        energy_calib.aligned_energy = float(split_line[aligned_index + 2])
+
+        energy_calib.energy_type = convert_energy_type(split_line[7])
+        energy_calib.operation = split_line[8]
+
+        if energy_calib.operation == "ADD":
+            energy_calib.energy_offset = (
+                min(energy_calib.measured_energies) - energy_calib.aligned_energy
+            )
+            energy_calib.operation = "addition"
+
+        energy_calib.validate_types()
+
+        self.casa_data["energy_calibrations"] += [energy_calib]
+
+    def process_intensity_calibration(self, calib_str: str):
+        """Process different intensity calibrations."""
+
+        intensity_calib = CasaIntensityCalibration()
+
+        def get_power_law_coefficient(line: str):
+            intensity_calib.power_law_coefficient = float(line.split("KECalib ")[1])
+
+        def get_transmission_function_ordinate(line: str):
+            intensity_calib.tf_function_ordinate_no = int(
+                line.split("TransmissionSet 1.0 ")[1]
+            )
+
+        def get_both_calibrations(line: str):
+            split_line = line.split(" ")
+            intensity_calib.power_law_coefficient = float(split_line[1])
+            intensity_calib.tf_function_ordinate_no = int(split_line[2])
+
+        calib_map = {
+            "KECalib": get_power_law_coefficient,
+            "TransmissionSet": get_transmission_function_ordinate,
+            "InCalib": get_both_calibrations,
+        }
 
```

Comment:
You can define in class level as this function works only for single string. The same dict will be created for multiple time.